File: src/main/kotlin/me/sd_master92/customvoting/gui/LuckyRewards.java
package me.sd_master92.customvoting.gui;

import me.sd_master92.customvoting.Main;
import me.sd_master92.customvoting.constants.Data;
import me.sd_master92.customvoting.constants.enumerations.SoundType;
import org.bukkit.ChatColor;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.InventoryClickEvent;
import org.bukkit.event.inventory.InventoryCloseEvent;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.jetbrains.annotations.NotNull;

public class LuckyRewards extends GUI
{
    public static final String NAME = "Lucky Rewards";

    public LuckyRewards(Main plugin)
    {
        super(plugin, NAME, 27, true);

        for (ItemStack reward : plugin.getData().getItems(Data.LUCKY_REWARDS))
        {
            getInventory().addItem(reward);
        }

        getInventory().setItem(25, BACK_ITEM);
        getInventory().setItem(26, SAVE_ITEM);
    }

    public static void save(Main plugin, Player player, Inventory inv)
    {
        inv.setItem(25, null);
        inv.setItem(26, null);
        if (plugin.getData().setItems(Data.LUCKY_REWARDS, inv.getContents()))
        {
            SoundType.SUCCESS.play(plugin, player);
            player.sendMessage(ChatColor.GREEN + "Successfully updated the Lucky Rewards!");
        } else
        {
            SoundType.FAILURE.play(plugin, player);
            player.sendMessage(ChatColor.RED + "Failed to update the Lucky Rewards!");
        }
    }

    @Override
    public void onClick(InventoryClickEvent event, Player player, @NotNull ItemStack item)
    {
        if (event.getSlot() >= 25)
        {
            event.setCancelled(true);
            if (event.getSlot() == 26)
            {
                LuckyRewards.save(plugin, player, event.getInventory());
            } else
            {
                SoundType.CLICK.play(plugin, player);
            }
            cancelCloseEvent();
            player.openInventory(new RewardSettings(plugin).getInventory());
        }
    }

    @Override
    public void onClose(InventoryCloseEvent event, Player player)
    {
        LuckyRewards.save(plugin, player, event.getInventory());
    }
}

File: src/main/kotlin/me/sd_master92/customvoting/gui/rewards/LuckyRewards.kt
package me.sd_master92.customvoting.gui.rewards

import me.sd_master92.core.inventory.GUI
import me.sd_master92.customvoting.CV
import me.sd_master92.customvoting.constants.enumerations.Data
import me.sd_master92.customvoting.constants.enumerations.SoundType
import me.sd_master92.customvoting.constants.enumerations.Strings
import org.bukkit.entity.Player
import org.bukkit.event.inventory.InventoryClickEvent
import org.bukkit.event.inventory.InventoryCloseEvent
import org.bukkit.inventory.Inventory
import org.bukkit.inventory.ItemStack

class LuckyRewards(private val plugin: CV) : GUI(plugin, name, 27, false)
{
    override fun onClick(event: InventoryClickEvent, player: Player, item: ItemStack)
    {
        if (event.slot >= 25)
        {
            event.isCancelled = true
            if (event.slot == 26)
            {
                save(plugin, player, event.inventory)
            } else
            {
                SoundType.CLICK.play(plugin, player)
            }
            cancelCloseEvent = true
            player.openInventory(RewardSettings(plugin).inventory)
        }
    }

    override fun onClose(event: InventoryCloseEvent, player: Player)
    {
        save(plugin, player, event.inventory)
    }

    companion object
    {
        private val name = Strings.GUI_TITLE_LUCKY_REWARDS.toString()
        fun save(plugin: CV, player: Player, inv: Inventory)
        {
            inv.setItem(25, null)
            inv.setItem(26, null)
            if (plugin.data.getItems(Data.LUCKY_REWARDS.path)
                    .contentEquals(inv.contents.filterNotNull().toTypedArray())
            )
            {
                SoundType.SUCCESS.play(plugin, player)
                player.sendMessage(Strings.UPDATE_NOTHING_CHANGED.toString())
            } else if (plugin.data.setItems(Data.LUCKY_REWARDS.path, inv.contents))
            {
                SoundType.SUCCESS.play(plugin, player)
                player.sendMessage(Strings.UPDATE_SUCCESS_X.with(name))
            } else
            {
                SoundType.FAILURE.play(plugin, player)
                player.sendMessage(Strings.UPDATE_FAIL_X.with(name))
            }
        }
    }

    init
    {
        for (reward in plugin.data.getItems(Data.LUCKY_REWARDS.path))
        {
            inventory.addItem(reward)
        }
        inventory.setItem(25, BACK_ITEM)
        inventory.setItem(26, SAVE_ITEM)
    }
}